watchdog/src/health.rs

```diff
@@ -0,0 +1,204 @@
+use crate::bitcoin_block_apis::BitcoinBlockApi;
+use crate::fetch::BlockInfo;
+use candid::CandidType;
+use serde::{Deserialize, Serialize};
+
+/// Status codes for the health of Bitcoin canister.
+#[derive(Clone, Debug, CandidType, PartialEq, Eq, Serialize, Deserialize)]
+pub enum StatusCode {
+    #[serde(rename = "no_data")]
+    NoData,
+    #[serde(rename = "ok")]
+    Ok,
+    #[serde(rename = "ahead")]
+    Ahead,
+    #[serde(rename = "behind")]
+    Behind,
+}
+
+/// The health status of the Bitcoin canister.
+#[derive(Clone, Debug, CandidType, PartialEq, Eq, Serialize, Deserialize)]
+pub struct HealthStatus {
+    pub source_height: Option<u64>,
+    pub other_number: u64,
+    pub other_heights: Vec<u64>,
+    pub target_height: Option<u64>,
+    pub height_diff: Option<i64>,
+    pub status: StatusCode,
+}
+
+/// Calculates the health status of the Bitcoin canister.
+pub fn calculate() -> HealthStatus {
+    compare(
```

Comment:
Not sure having the `compare` as an extra function provides a lot of benefit. I'd keep only `calculate` (and maybe rename to `get_health_status`).

Reply:
A separate `compare` is needed to easily unit test different cases without getting into storage read/write.